src/materials/EEDFRateConstantTownsend.C

```diff
@@ -58,32 +56,10 @@ EEDFRateConstantTownsend::EEDFRateConstantTownsend(const InputParameters & param
     _em(coupledValue("electrons")),
     _mean_en(coupledValue("mean_energy"))
 {
-  std::vector<Real> temp_x;
-  std::vector<Real> temp_y;
-  std::vector<Real> actual_mean_energy;
-  std::vector<Real> rate_coefficient;
-  std::string file_name =
-      getParam<std::string>("file_location") + "/" + getParam<FileName>("property_file");
-  MooseUtils::checkFileReadable(file_name);
-  const char * charPath = file_name.c_str();
-  std::ifstream myfile(charPath);
-  Real value;
-
-  if (myfile.is_open())
-  {
-    while (myfile >> value)
-    {
-      temp_x.push_back(value);
-      myfile >> value;
-      temp_y.push_back(value);
-
-      actual_mean_energy.push_back(1);
-      rate_coefficient.push_back(1);
-    }
-    myfile.close();
-  }
-  else
-    mooseError("Unable to open file");
+  const auto [temp_x, temp_y] = CraneUtils::getReactionRates(*this); // just need the sizes
```

Comment:
Mac testing is failing to build because of the structured binding here, saying that capturing `temp_x` cannot be done in Line 69:
```
std::sort(
      idx.begin(), idx.end(), [&temp_x](size_t i1, size_t i2) { return temp_x[i1] < temp_x[i2]; });
```

This is formally allowed by the standard in C++20 (and C++17 with GCC it seems, since Linux testing is happy) but doesn't appear to be happy with Clang.

Reply:
Should be good now.